pkg/controllers/namespace/namespace_controller.go

```diff
@@ -96,8 +89,10 @@ func NewNamespaceController(k8sClientset *kubernetes.Clientset, calicoClient *cl
 				log.Errorf("Error while converting %#v to calico profile.", obj)
 				return
 			}
-			// Add profileName:*profile in calicoCache
-			ccache.Set(profile.(api.Profile).Metadata.Name, profile)
+
+			calicoKey := namespaceConverter.GetKey(profile)
+			// Add key:*profile in calicoCache
```

Comment:
@svInfra17 I think you missed the one that this comment was on :)

Others look good though.